src/compiler/checker.ts

```diff
@@ -21876,6 +21878,9 @@ namespace ts {
             checkGrammarDecoratorsAndModifiers(node);
 
             checkVariableLikeDeclaration(node);
+            if (node.name && isIdentifier(node.name) && node.name.originalKeywordKind === SyntaxKind.PrivateName) {
+                error(node, Diagnostics.Private_names_cannot_be_used_as_parameters);

```

Comment:
we plan to undo this change is in https://github.com/bloomberg/TypeScript/pull/5